activerecord/test/cases/adapters/postgresql/postgresql_adapter_test.rb

```diff
@@ -481,6 +481,7 @@ def test_unparsed_defaults_are_at_least_set_when_saving
 
           first_number.save!
           assert_equal 4, first_number.reload.number
+          @connection.schema_cache.clear_data_source_cache!(number_klass.table_name)
```

Comment:
Why only those two tests needs to clean the schema cache given this table is used in more places than here?

Reply:
We were thinking about that too e.g. does this need to happen for all the tests that use the same table name but we wanted to start by addressing this failure case. Should we do this for every test using the same table?

It feels weird to have to explicitly clear the schema cache to begin with. We were expecting the schema cache to be automatically cleared whenever a table is dropped.

Reply:
An alternative we discussed would be to only clear the schema cache in the failing test just before table creation.

Reply:
I'd investigate why the schema cache is keeping this information in this test only. If it is always keeping the information and other tests aren't failing because they are not exercising this codepath, we should make a generic "clean the cache after tests on this table" otherwise other tests could be added later that would fail for the same reason and we would have to investigate it again.

Reply:
We looked into the lifecycle of the schema cache during test run, and we can see (using debug print) that the `schema_cache` is instantiated upon connection creation. The information builds up in the `schema_cache` as and when calls are made to retrieve column information e.g. `columns_hash`, `column_for_attribute`. The `schema_cache` is never cleared between tests. Looking at `DdlHelper` which creates table via `with_example_table`, we can see that tables are dropped by executing a raw query without calling the clear method on the schema cache. Interestingly, the `SchemaStatements` module clears the schema cache as part of the `drop_table` method.
So, I think it makes sense to go with the generic "clean the cache after tests on this table".